src/outfit.ts

```diff
@@ -14,67 +14,79 @@ import { $familiar, $item, $skill, $slot, $slots, get, have, Requirement } from
-    if (Array.isArray(item)) return item.every((val) => this.equip(val));
-    if (!have(item)) return false;
-    if (item instanceof Item && !canEquip(item)) return false;
-    if (this.avoid && this.avoid.find((i) => i === item) !== undefined) return false;
+  private equipItem(item: Item): boolean {
+    if (!have(item) || !canEquip(item)) return false;
+    if (this.avoid?.find((i) => i === item)) return false;
 
-    if (item instanceof Item) {
-      const slot = toSlot(item);
-      if (slot === $slot`acc1`) {
-        if (this.accesories.length >= 3) return false;
-        this.accesories.push(item);
+    const slot = toSlot(item);
+    switch (slot) {
+      case $slot`acc1`:
+        if (this.accessories.length >= 3) return false;
+        this.accessories.push(item);
         return true;
-      }
-      if (slot === $slot`off-hand`) {
-        const weapon = this.equips.get($slot`weapon`);
-        if (weapon && weaponHands(weapon) === 2) return false;
-      }
-      if (!this.equips.has(slot)) {
-        this.equips.set(slot, item);
-        return true;
-      }
-      if (
-        slot === $slot`weapon` &&
-        !this.equips.has($slot`off-hand`) &&
-        have($skill`Double-Fisted Skull Smashing`) &&
-        weaponHands(item)
-      ) {
-        this.equips.set($slot`off-hand`, item);
-        return true;
-      }
-      if (
-        slot === $slot`off-hand` &&
-        have($familiar`Left-Hand Man`) &&
-        this.familiar === undefined &&
-        !this.equips.has($slot`familiar`)
-      ) {
-        // eslint-disable-next-line libram/verify-constants
-        if (item === $item`cursed magnifying glass` && !canChargeVoid()) {
-          // Cursed magnifying glass cannot trigger in Lefty
-          this.equips.set($slot`familiar`, this.equips.get($slot`off-hand`) ?? $item`none`);
+
+      case $slot`off-hand`:
+        if (
+          weaponHands(this.equips.get($slot`weapon`) ?? $item`none`) === 2 ||
+          this.equips.has($slot`off-hand`)
+        ) {
+          if (
+            have($familiar`Left-Hand Man`) &&
+            [undefined, $familiar`Left-Hand Man`].includes(this.familiar) &&
+            !this.equips.get($slot`familiar`)
+          ) {
+            if (item === $item`cursed magnifying glass` && !canChargeVoid()) {
+              const current = this.equips.get($slot`off-hand`);
+              this.equips.set($slot`off-hand`, item);
+              current
+                ? this.equips.set($slot`familiar`, current)
+                : this.equips.delete($slot`familiar`);
+              return true;
+            }
+            this.equips.set($slot`familiar`, item);
+            this.familiar = $familiar`Left-Hand Man`;
+            return true;
+          }
+          return false;
+        }
+        break;
+
+      case $slot`weapon`:
+        if (
+          this.equips.has($slot`weapon`) &&
+          have($skill`Double-Fisted Skull Smashing`) &&
+          weaponHands(item) === 1
```

Comment:
And no current off-hand

Reply:
And, actually, that the current weapon is only 1-handed. Which isn't checked in the original code either.

Reply:
update I did that but also that might be worse

Reply:
I think we still need to check here that the off-hand is currently empty before slotting a weapon into it.

Reply:
update: finally actually checked the handedness of current weapon, whoops